Applications/Spire/Source/Ui/TextBox.cpp

```diff
@@ -98,7 +118,58 @@ QSize TextBox::sizeHint() const {
 }
 
 void TextBox::style_updated() {
+  auto style = get_style();
+  auto placeholder_style = QString("QLabel {");
```

Comment:
This needs to be done in the ```paintEvent``` since the parent can override a child's style. You can check whether or not the placeholder is being displayed and if it is then style it.

Reply:
It’s tricky. In the TextBox, there are two components: a line edit and a label. The label covers the line edit. The computed style is assigned to the line edit in paintEvent. The computed style combines different styles in different states, but here needs two different styles for the line edit and the label. For example, the TextBox and placeholder may have two different background colors (we don’t have this requirement now), but the computed style only provides one background color for these two components. For the text color, the inheritance can solve the issue, because the input text and the placeholder text don’t appear at the same time and the line edit and the label can share the text color. Another question is that the font is set by the setFont function. The computed style provides one font object, so we don’t know which component should be set with the font.